src/components/Nav/index.tsx

```diff
@@ -433,20 +415,28 @@ const Nav: React.FC<IProps> = ({ path }) => {
             </LeftItems>
             <RightItems>
               <Search useKeyboardShortcut />
-              <ThemeToggle
-                onClick={toggleColorMode}
+              <IconButton
                 aria-label={
                   isDarkTheme ? "Switch to Light Theme" : "Switch to Dark Theme"
                 }
-              >
-                <NavIcon name={isDarkTheme ? "lightTheme" : "darkTheme"} />
-              </ThemeToggle>
-              <RightNavLink to="/languages/">
-                <NavIcon name="language" />
-                <Span>
-                  <Translation id="languages" />
-                </Span>
-              </RightNavLink>
+                icon={
+                  <Icon
+                    as={isDarkTheme ? MdWbSunny : MdBrightness2}
+                    fontSize="2xl"
+                  />
+                }
+                variant="icon"
+                _hover={{ color: "primary" }}
+                onClick={toggleColorMode}
+              />
+              <ButtonLink to="/languages/" variant="icon">
+                <Flex flexWrap="nowrap" alignItems="center">
```

Comment:
It is not semantically correct to place a div inside the button. I've removed it and is not wrapping.
```suggestion
```